VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/select-hearing-layout/select-hearing-layout.component.spec.ts

```diff
@@ -34,20 +37,44 @@ describe('SelectHearingLayoutComponent', () => {
         videoCallService.getPreferredLayout.and.returnValue(layout);
         component.ngOnInit();
         expect(component.selectedLayout).toBe(layout);
+        expect(component.currentButtonContentKey).toBe(buttonContentKeyWhenOpen);
+    });
+
+    it('should show translated text on open accordion button', () => {
+        const expectedTranslatedContentForButton = 'this is translated for open all button';
+        component.currentButtonContentKey = buttonContentKeyWhenOpen;
+        translateServiceSpy.instant
+            .withArgs(`select-hearing-layout.${component.currentButtonContentKey}`)
+            .and.returnValue(expectedTranslatedContentForButton);
+        component.ngOnInit();
+        onLangChangeSpy.emit({ lang: 'tl' } as LangChangeEvent);
+        expect(component.accordionOpenAllElement.innerHTML).toContain(expectedTranslatedContentForButton);
+    });
+
+    it('should show translated text on open/close toggle button', () => {
+        const expectedTranslatedContentForHeader = 'this is translated for the accordion header';
+        translateServiceSpy.instant
+            .withArgs('select-hearing-layout.choose-hearing-layout')
+            .and.returnValue(expectedTranslatedContentForHeader);
+        component.ngOnInit();
+        onLangChangeSpy.emit({ lang: 'tl' } as LangChangeEvent);
+        expect(headingButton.innerHTML).toContain(expectedTranslatedContentForHeader);
     });
 
     it('should translate button text on text click', () => {
         component.ngOnInit();
         textButton.innerHTML = 'Close all';
         textButton.click();
         expect(textButton.innerHTML).toBe('<span>select-hearing-layout.close-all</span>');
+        expect(component.currentButtonContentKey).toBe(buttonContentKeyWhenClosed);
```

Comment:
I think it would be better to use spyOn(component, 'setAccordionText') and expect that it is called correctly.

Reply:
should I make this change as it was an existing test, I added that assertion as I introduced that new property for this component? However it is tested in multiple places